src/reference/antora/modules/ROOT/pages/cloudevents-transform.adoc

```diff
@@ -0,0 +1,133 @@
+[[cloudevents-transformer]]
+
+= CloudEvent Transformer
+
+[[introduction]]
+== Introduction
+
+The CloudEvent transformer converts Spring Integration messages into CloudEvent compliant messages.
+This transformer provides  support for the CloudEvents specification v1.0 with configurable output format and defining attributes and extensions using `Expression`s.
+
+[[cloudevent-transformer-overview]]
+=== Overview
+
+The CloudEvent transformer (`ToCloudEventTransformer`) extends Spring Integration's `AbstractTransformer` to convert messages to CloudEvent format.
+The CloudEvent transformer identifies the `EventFormat` classes in the classpath and utilizes registers these as the available serializers for CloudEvents.
+The type of serialization (JSON, XML, AVRO, etc) of the CloudEvent message is determined by `contentType` of the message.
+
+NOTE: Messages to be transformed must have a payload of `byte[]`.
+
+
+
+[[configure-transformer]]
+=== Configuring Transformer
+
+The `ToCloudEventTransformer` allows the user to use SpEL `Expression`s to populate the attributes as well as the extensions.
+
+==== Attribute Expressions
+As discussed above users are allowed to set the `id`, `source`, `type`, `dataSchema`, `subject` through SpEL `Expression`s.
+The example below shows where a `ToCloudEventTransformer` is created with a null `expression`s variable.
+This indicates that this transformer will not place any `extensions` in the CloudEvent.
+But the user does want to set the type of the CloudEvent to `sampleType`.
+
+NOTE: The `time` attribute is set to the time that the CloudEvent message was created by the `ToCloudEventTransformer` transformer.
+
+[source,java]
+----
+ExpressionParser parser = new SpelExpressionParser();
+ToCloudEventTransformer transformer = new ToCloudEventTransformer(null);
+transformer.setTypeExpression(parser.parseExpression("sampleType"));
+----
+
+==== Extension Expressions
+The expressions constructor parameter is an array of `Expression`s.
```

Comment:
Again: blank line after the title.
And probably this section would be changed if we come back to patterns for headers to map into extensions.
Unless you can convince me why we have to go with expressions for them.